CMakeLists.txt

```diff
@@ -29,15 +31,32 @@ project(reflectcpp)
 
 set(CMAKE_CXX_STANDARD 20)
 
-if (REFLECTCPP_BUILD_SHARED)
-    add_library(reflectcpp SHARED src/yyjson.c)
-else ()
-    add_library(reflectcpp STATIC src/yyjson.c)
-endif ()
+if(REFLECTCPP_USE_BUNDLED_DEPENDENCIES)
+    if (REFLECTCPP_BUILD_SHARED)
+        add_library(reflectcpp SHARED src/yyjson.c)
+    else()
+        add_library(reflectcpp STATIC src/yyjson.c)
+    endif()
+else()
+    if (REFLECTCPP_BUILD_SHARED)
+        add_library(reflectcpp SHARED)
+    else()
+        add_library(reflectcpp STATIC)
+    endif()
+endif()
 
+set_target_properties(reflectcpp PROPERTIES LINKER_LANGUAGE CXX)
```

Comment:
Instead, probably could use the `LANGUAGES` section of `project`
https://cmake.org/cmake/help/latest/command/project.html

But setting it just to `CXX`, as this is a C++ project, gives trouble with the yyjson.c .
I guess it could be a C++ wrapper around it. 
Maybe like 
```cpp
// yyjson.cpp
extern C {
  #include "yyjson.c"
}
```

Quick test shows, that at least clangd is happy with it, and uses the right compiler and flags.